### PR TITLE
Extended Syntax: constant propagation

### PR DESCRIPTION
**Pull request by Anabra**
See #32

### CODE REVIEW

File: grin/src/Transformations/ExtendedSyntax/Optimising/ConstantPropagation.hs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+-}
+
+type Env = Map Name Val
+
+constantPropagation :: Exp -> Exp
+constantPropagation e = ana builder (mempty, e) where
+
+  builder :: (Env, Exp) -> ExpF (Env, Exp)
+  builder (env, exp) = case exp of
+    ECase scrut alts -> -- val ~ scrut
```
Comment (andorp) [inline grin/src/Transformations/ExtendedSyntax/Optimising/ConstantPropagation.hs:31]: Nit: Outdated comment.

File: grin/src/Transformations/ExtendedSyntax/Optimising/ConstantPropagation.hs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      let constVal      = getValue scrut env -- error "substValsVal env val"
```
Comment (andorp) [inline grin/src/Transformations/ExtendedSyntax/Optimising/ConstantPropagation.hs:32]: Nit: outdated comment.

File: grin/src/Transformations/ExtendedSyntax/Optimising/ConstantPropagation.hs (hunk 1 above)
Comment (Anabra) [inline grin/src/Transformations/ExtendedSyntax/Optimising/ConstantPropagation.hs:31]:
```suggestion
    ECase scrut alts ->
```

File: grin/src/Transformations/ExtendedSyntax/Optimising/ConstantPropagation.hs (hunk 2 above)
Comment (Anabra) [inline grin/src/Transformations/ExtendedSyntax/Optimising/ConstantPropagation.hs:32]:
```suggestion
      let constVal      = getValue scrut env 
```

**Pull request merged**